fedimint-client/src/lib.rs

```diff
@@ -1919,6 +1920,70 @@ impl Client {
             Result::<_, ()>::Ok(guardian_pub_keys)
         }), None).await.expect("Will retry forever")
     }
+
+    pub fn handle_global_rpc(
+        &self,
+        method: String,
+        params: serde_json::Value,
+    ) -> BoxStream<'_, anyhow::Result<serde_json::Value>> {
+        Box::pin(try_stream! {
+            match method.as_str() {
+                "get_balance" => {
+                    let balance = self.get_balance().await;
+                    yield serde_json::to_value(balance)?;
+                }
+                "subscribe_balance_changes" => {
+                    let mut stream = self.subscribe_balance_changes().await;
+                    while let Some(balance) = stream.next().await {
+                        yield serde_json::to_value(balance)?;
+                    }
+                }
```

Comment:
Won't this just block the stream forever, not allowing any other commands to be processed? Or is the idea to open new streams for such commands anyway?

Reply:
yes streams will concurrently run.

https://github.com/fedimint/fedimint/pull/5595/files#diff-df844475f10253d85671819b33bad8d7ae1fcd5da5eb1f3c773ece59a874cd42R72

javascript can call rpc function even when other rpc is not finished yet.